### PR TITLE
update(JS): web/javascript/reference/global_objects/promise

### PR DESCRIPTION
**Pull request by undead404**
Оригінальний вміст: [Promise@MDN](https://developer.mozilla.org/en-us/docs/Web/JavaScript/Reference/Global_Objects/Promise), [сирці Promise@GitHub](https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/promise/index.md)

Нові зміни:
- [Update redirected links, part1 (#35216)](https://github.com/mdn/content/commit/e03b13c7e157ec7b7bb02a6c7c4854b862195905)
- [Synchronize with BCD v5.5.42 (#35175)](https://github.com/mdn/content/commit/8b920a5e7567dcc9d642dfbd704b0ddbe2005d30)
- [Convert noteblocks for web/javascript/reference/global_objects folder (#35091)](https://github.com/mdn/content/commit/8421c0cd94fa5aa237c833ac6d24885edbc7d721)
- [Reference for stage 3 promise-try (#34929)](https://github.com/mdn/content/commit/65151916d0bd24bcb2dab686825b65ca84600d7c)
- [Remove all `@@notation` from page slugs (#34824)](https://github.com/mdn/content/commit/6fbdb78c1362fae31fbd545f4b2d9c51987a6bca)
- [Remove all @@notation in JS prose (#34817)](https://github.com/mdn/content/commit/6e93ec8fc9e1f3bd83bf2f77e84e1a39637734f8)
- [promise: see also AbortController (#33346)](https://github.com/mdn/content/commit/79d0563ccd318fa5c3c05d74633c7fa802db2db3)
- [#33742 Fix analogy for chained promises (#33790)](https://github.com/mdn/content/commit/9f48a53644b6b77b555f10cf6d6e829e1d9531f0)

### CODE REVIEW

File: files/uk/web/javascript/reference/global_objects/promise/index.md [hunk 1]
```diff
@@ -63,9 +63,21 @@ myPromise
...
 
-Обробка відхиленого промісу в кожному `.then()` має наслідки далі всередині ланцюжка промісів. Інколи просто немає вибору, оскільки помилка має бути опрацьована одразу. В такому випадку ми повинні викинути помилку якогось типу, щоб зберегти вміст помилки далі всередині послідовності. З іншого боку, в разі відсутності нагальної потреби простіше віддати обробку помилки останній інструкції `.catch()` у ланцюжку. Фактично `.catch()` — це просто `.then()`, тільки без гнізда для функції успішного сповнення промісу.
+Наприклад, у коді вище, якщо `myPromise` відхиляється, то викликається `handleRejectedA`, і якщо `handleRejectedA` завершиться нормально (без викидання помилки чи повернення відхиленого проміса), то проміс, повернений першим `then`, буде сповнений, а не залишиться відхиленим. Тому, якщо помилку треба обробити негайно, але потрібно зберегти стан помилки для використання в ланцюжку, необхідно викинути помилку певного типу у обробнику відхилення. З іншого боку, якщо немає негайної потреби, простіше покласти обробку помилок на останній обробник `catch()`.
```
[the remark this answers is not in the document]
Comment (undead404) [reply files/uk/web/javascript/reference/global_objects/promise/index.md:80]:
```suggestion
Наприклад, у коді вище, якщо `myPromise` відхиляється, то викликається `handleRejectedA`, і якщо `handleRejectedA` завершиться нормально (без викидання помилки чи повернення відхиленого проміса), то проміс, повернений першим `then`, буде сповнений, а не залишиться відхиленим. Тому, якщо помилку треба обробити негайно, але потрібно зберегти стан помилки для використання в ланцюжку, необхідно викинути помилку певного типу в обробнику відхилення. З іншого боку, якщо немає негайної потреби, простіше покласти обробку помилок на останній обробник `catch()`.
```

File: files/uk/web/javascript/reference/global_objects/promise/index.md [hunk 2]
```diff
@@ -101,29 +113,17 @@ myPromise
...
 
 ```js
 const promiseA = new Promise(myExecutorFunc);
 const promiseB = promiseA.then(handleFulfilled1, handleRejected1);
 const promiseC = promiseA.then(handleFulfilled2, handleRejected2);
 ```
 
-Можна прив'язати якусь дію до вже "залагодженого" промісу. В цьому випадку дію (якщо це допустимо) буде виконано при першій ліпшій асинхронній можливості. Зауважте, що проміси гарантовано асинхронні. Таким чином, дія для уже "залагодженого" промісу трапиться лише після того, як стек викликів буде очищено і мине мить таймера рушія. Цей ефект дуже схожий до результату виконання `setTimeout(action, 0)`.
+Можна прив'язати якусь дію до вже залагодженого промісу. У цьому випадку дія додається в кінець черги завдань негайно, і виконується, коли завершуютьсяя всі наявні завдання. Таким чином, дія для вже "залагодженого" промісу відбудеться лише після завершення поточного синхронного коду та принаймні одного циклу обробки. Це створює гарантію того, що дії промісів – асинхронні.
```
[the remark this answers is not in the document]
Comment (undead404) [reply files/uk/web/javascript/reference/global_objects/promise/index.md:126]:
```suggestion
Можна прив'язати якусь дію до вже залагодженого промісу. У цьому випадку дія додається в кінець черги завдань негайно, і виконується, коли завершуються всі наявні завдання. Таким чином, дія для вже "залагодженого" промісу відбудеться лише після завершення поточного синхронного коду та принаймні одного циклу обробки. Це створює гарантію того, що дії промісів – асинхронні.
```

File: files/uk/web/javascript/reference/global_objects/promise/index.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ browser-compat: javascript.builtins.Promise
 
 ## Опис
 
-**`Promise`** – це заміна для значення, яке, як правило, невідоме на момент, коли проміс створюється. Він дає змогу прив'язати обробники до результату успішного виконання чи до причини невдачі асинхронної операції. Це дає асинхронним методам змогу повертати значення схоже до того, як це роблять синхронні функції. Замість повернути одразу кінцеве значення метод повертає _promise - зобов'язання_ надати це значення в якийсь момент у майбутньому.
+`Promise` – це заміна для значення, яке, як правило, невідоме на момент, коли проміс створюється. Він дає змогу прив'язати обробники до результату успішного виконання чи до причини невдачі асинхронної операції. Це дає асинхронним методам змогу повертати значення схоже до того, як це роблять синхронні функції. Замість повернути одразу кінцеве значення метод повертає _promise - зобов'язання_ надати це значення в якийсь момент у майбутньому.
```
[the remark this answers is not in the document]
Comment (undead404) [reply files/uk/web/javascript/reference/global_objects/promise/index.md:16]: @alinkedd ось іще одна кумедна ситуація

[the remark this answers is not in the document]
Comment (alinkedd) [reply files/uk/web/javascript/reference/global_objects/promise/index.md:16]: same
https://github.com/webdoky/content/pull/3242#discussion_r1709215540

File: files/uk/web/javascript/reference/global_objects/promise/index.md [hunk 4]
```diff
@@ -99,31 +112,20 @@ myPromise
...
 
 ```js
 const promiseA = new Promise(myExecutorFunc);
 const promiseB = promiseA.then(handleFulfilled1, handleRejected1);
 const promiseC = promiseA.then(handleFulfilled2, handleRejected2);
 ```
 
-Можна прив'язати якусь дію до вже "залагодженого" промісу. В цьому випадку дію (якщо це допустимо) буде виконано при першій ліпшій асинхронній можливості. Зауважте, що проміси гарантовано асинхронні. Таким чином, дія для уже "залагодженого" промісу трапиться лише після того, як стек викликів буде очищено і мине мить таймера рушія. Цей ефект дуже схожий до результату виконання `setTimeout(action, 0)`.
+Можна прив'язати якусь дію до вже залагодженого промісу. У цьому випадку дія додається в кінець черги завдань негайно, і виконується, коли завершуються всі наявні завдання. Таким чином, дія для вже "залагодженого" промісу відбудеться лише після завершення поточного синхронного коду та принаймні одного циклу обробки. Це створює гарантію того, що дії промісів – асинхронні.
```
Comment (alinkedd) [inline files/uk/web/javascript/reference/global_objects/promise/index.md:128]:
```suggestion
Можна прив'язати якусь дію до вже залагодженого промісу. У цьому випадку дія додається в кінець черги завдань негайно і виконується, коли завершуються всі наявні завдання. У такий спосіб дія для вже "залагодженого" промісу відбудеться лише після завершення поточного синхронного коду та принаймні одного циклу обробки. Це створює гарантію того, що дії промісів – асинхронні.
```

**Review by alinkedd — CHANGES_REQUESTED** [review]

File: files/uk/web/javascript/reference/global_objects/promise/index.md [hunk 5]
```diff
@@ -42,13 +42,14 @@ new Promise((resolveOuter) => {
...
+### Ланцюжки промісів
 
-Метод `.then()` приймає до двох аргументів; перший — це функція зворотного виклику для сповнення промісу, а другий — відповідно, функція для його відхилення. Кожний `.then()` повертає новостворений об'єкт промісу, який далі може (необов'язково) використовуватися для нарощування ланцюжка викликів, як от:
+Методи промісів {{jsxref("Promise/then", "then()")}}, {{jsxref("Promise/catch", "catch()")}} і {{jsxref("Promise/finally", "finally()")}} використовуються, щоб приєднати якусь наступну дію до промісу, який залагоджується. Метод `.then()` приймає до двох аргументів; перший — це функція зворотного виклику на випадок сповнення промісу, а другий — відповідно, на випадок його відхилення. Методи `catch()` і `finally()` за лаштунками викликають `then()` і роблять обробку помилок менш громіздкою. Наприклад, насправді `catch()` – це просто `then()` без передачі обробника сповнення. Оскільки ці методи повертають проміси, їхні виклики можна об'єднувати в ланцюжки. Наприклад:
```
[the remark this answers is not in the document]
Comment (undead404) [reply files/uk/web/javascript/reference/global_objects/promise/index.md:52]: Може й так, але я таке прямо перекладаю, тому fulfillment – сповнення

[the remark this answers is not in the document]
Comment (alinkedd) [reply files/uk/web/javascript/reference/global_objects/promise/index.md:52]: ок

**Review by alinkedd — APPROVED** [review]

**Pull request merged**